crates/iota-grpc-api/src/client.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_grpc_types::{CertifiedCheckpointSummary, CheckpointData};
+use tonic::transport::Channel;
+
+use crate::node::node_service_client::NodeServiceClient;
+
+/// Enum representing the content of a checkpoint, either full data or summary.
+pub enum CheckpointContent {
+    Data(CheckpointData),
+    Summary(CertifiedCheckpointSummary),
+}
+/// Shared gRPC client for IOTA node operations.
+pub struct GrpcNodeClient {
+    client: NodeServiceClient<Channel>,
+}
+
+impl GrpcNodeClient {
+    pub async fn connect(url: &str) -> Result<Self, tonic::transport::Error> {
+        let client = NodeServiceClient::connect(url.to_string()).await?;
+        Ok(Self { client })
+    }
+
+    /// Stream checkpoints with any combination of start and end sequence
+    /// numbers.
+    pub async fn stream_checkpoints(
+        &mut self,
+        start_sequence_number: Option<u64>,
+        end_sequence_number: Option<u64>,
+        full: Option<bool>,
+    ) -> Result<tonic::Streaming<crate::node::Checkpoint>, tonic::Status> {
```

Comment:
Since this is the client, we could directly return the "bcs deserialized" types IMO. This would remove the whole `deserialize_checkpoint` step for the user.